AWS/sagemaker/pytorch_utils/models/conditional_vae.py

```diff
@@ -0,0 +1,150 @@
+			.. hidden_layers (List[int]) - denotes length dimension of hidden layers' weight. 
+			.. num_classes (int) - number of classes/target groups in dataset
+		"""
+		super().__init__()
+
+		if not isinstance(hidden_layers, (list, tuple)) or len(hidden_layers) < 1:
+			raise ValueError('Error: CEncoder is initialised without hidden layers.')
+
+		self.input_dim = input_dim
+		self.latent_dim = latent_dim
+		self.num_classes = num_classes
+
+		# Initialise first hidden layer
+		self.hidden_layers = nn.ModuleList([
+			nn.Linear(input_dim + num_classes, hidden_layers[0])
+		])
+
+		# Add more hidden layers
+		zipped_layers = zip(hidden_layers[:-1], hidden_layers[1:])
+		self.hidden_layers.extend([nn.Linear(h1, h2) for h1, h2 in zipped_layers])
+
+		# Bottle neck layers
+		self.mu_layer = nn.Linear(hidden_layers[-1], latent_dim)      # shape: [batch_size, latent_dim]
+		self.logvar_layer = nn.Linear(hidden_layers[-1], latent_dim)  # shape: [batch_size, latent_dim]
+
+
+	def forward(self, x):
+		# Input x shape: [batch_size, input_dim + num_classes]
+		x = x.view(-1, self.input_dim + self.num_classes)
+
+		# Forward flow to hidden layers
+		for linear in self.hidden_layers:
+			x = F.relu(linear(x))
+
+		# Latent space parametric variables
+		z_mu = self.mu_layer(x)
+		z_logvar = self.logvar_layer(x)
+
+		return z_mu, z_logvar
+
+
+class CDecoder(nn.Module):
+	""" Decoder module for Conditional VAE network
+	"""
+
+	def __init__(self, input_dim, latent_dim, hidden_layers, num_classes):
+		""" Initialise Decoder component of CVAE
+		Arguments:
+		------------------------
+			.. input_dim (int) - input channel of data X, the length of flattened image vector
+			.. latent_dim (int) - length of latent dim vector in bottle neck layer
+			.. hidden_layers (List[int]) - denotes length dimension of hidden layers' weight. 
+			.. num_classes (int) - number of classes/target groups in dataset
+		"""
+		super().__init__()
+
+		if not isinstance(hidden_layers, (list, tuple)) or len(hidden_layers) < 1:
+			raise ValueError('Error: CEncoder is initialised without hidden layers.')
+
+		self.input_dim = input_dim
+		self.latent_dim = latent_dim
+		self.num_classes = num_classes
+
+		# Initialise first hidden layer
+		self.hidden_layers = nn.ModuleList([
+			nn.Linear(latent_dim + num_classes, hidden_layers[0])
+		])
+
+		# Add more hidden layers
+		zipped_layers = zip(hidden_layers[:-1], hidden_layers[1:])
+		self.hidden_layers.extend([nn.Linear(h1, h2) for h1, h2 in zipped_layers])
+
+		# Recontruction layer
+		self.recontruction_layer = nn.Linear(hidden_layers[-1], input_dim)
+
+	def forward(self, z):
+		# Latent variable z shape: [batch_size, latent_dim + num_classes]
+		x = z.view(-1, self.latent_dim + self.num_classes)
+
+		# Forward flow to hidden layers
+		for linear in self.hidden_layers:
+			x = F.relu(linear(x))
+
+		# Recontructed input image
+		generated_x = F.sigmoid(self.recontruction_layer(x))
+
+		return generated_x
+
+
+class CVAE(nn.Module):
```

Comment:
Put this at the very start of the file above CEncoder and CDecoder

Reply:
noted. will do and re-run the script after changing so.